moshi/src/main/java/com/squareup/moshi/ClassJsonAdapter.java

```diff
@@ -75,6 +75,14 @@
       if (Modifier.isAbstract(rawType.getModifiers())) {
         throw new IllegalArgumentException("Cannot serialize abstract class " + rawType.getName());
       }
+      for (Annotation annotation : rawType.getDeclaredAnnotations()) {
```

Comment:
This is a very expensive reflection call. Calling isAnnotationPresent is much faster. Can you change to that?

Reply:
You might need to load the metadata class using Class.forName() and that can fail. If it does, I don't _think_ it's possible for Kotlin to be present.

Reply:
> Calling isAnnotationPresent is much faster. Can you change to that?

Can do, completely forgot about that. #754

> I don't think it's possible for Kotlin to be present

Kotlin technically could be present, but only if the stdlib isn't used. We wouldn't be able to tell in this case :/